### DISCUSSION TITLE
Is it Possible to load glb and make UI like this - Background Control , Forground Control

### DISCUSSION THREAD

**Discussion by atul161** [google/filament#6417] [General]
Hi @romainguy , @bejado , @pixelflinger  !

Thanks for building such a strong project.  Currently i want to build
UI like this. Here is the public drive link attached.

<img src="https://drive.google.com/uc?export=view&id=13cIA8JaUD0uv46S0OVkY_WGVP-ZzXxoL" width="320">

 But i am not able to find the way through i can achieve this.

Just Need team views how can i achieve this or it is even possible to achieve this or not using filament.

- I want to control the background image and on the top of background image my model should be placed.
- I dont want to optimize the quality of background image as well
- I also want my UI element to not overlap.
- I don't want to use texture view as this will cause some performance issue.

I am continously struggling with this and want to achieve this any help would be appreciated.

Hope I will get some answer from you guys!!

**Comment by romainguy** [comment]
You can use a SurfaceView for this and have the UI controls be regular Android views. You can draw the background in Filament itself. To get further help please ask more specific questions and we'll be happy to guide you.

**Reply by atul161** [reply]
Thanks !


Just to be clear my avatar is in glb format and background image  is  in png/jpeg format.

Currently i am not able to find a way so that i can put my glb above my background image.

**Reply by romainguy** [reply]
You can do this by adding a triangle or quad in the Filament scene, using the `device` [vertex domain](https://google.github.io/filament/Materials.html#materialdefinitions/materialblock/vertexandattributes:vertexdomain). If you set the Z values to 1 (or something very close to 1), that triangle/quad will always be behind your other objects. You can then apply a texture to that triangle/quad. That's how the skybox works for instance.

Here is a [C++ example](https://github.com/google/filament/blob/main/samples/image_viewer.cpp#L158) that does exactly this. The `createImageRenderable()` function creates the renderable as a full screen triangle that's always at Z=1 in device space.

**Reply by atul161** [reply]
Thankyou ! @romainguy for your help! I have successfully achieved this.:)
